docs/protocol.html

```diff
@@ -109,13 +109,13 @@
 
 <h5><a id="protocol_compatibility" href="#protocol_compatibility">Versioning and Compatibility</a></h5>
 
-<p>The protocol is designed to enable incremental evolution in a backward compatible fashion. Our versioning is on a per-api basis, each version consisting of a request and response pair. Each request contains an API key that identifies the API being invoked and a version number that indicates the format of the request and the expected format of the response.</p>
+<p>Kafka APIs are uniquely identified by numbers, called &#34;api keys.&#34;  For example, a ProduceRequest is identified by ApiKey 0.  A ListGroupsRequest is identified by ApiKey 16.  When adding a new API, we always assign it a new API key.  API keys are never reused.
 
-<p>The intention is that clients would implement a particular version of the protocol, and indicate this version in their requests. Our goal is primarily to allow API evolution in an environment where downtime is not allowed and clients and servers cannot all be changed at once.</p>
+<p>Each APIs is versioned.  The version numbers start at 0 and go up from there.  The version number determines the format of what is sent over the wire and how it is to be interpreted.
 
-<p>The server will reject requests with a version it does not support, and will always respond to the client with exactly the protocol format it expects based on the version it included in its request. The intended upgrade path is that new features would first be rolled out on the server (with the older clients not making use of them) and then as newer clients are deployed these new features would gradually be taken advantage of.</p>
+<p>In order to provide cross-version compatibility, brokers can handle a range of versions.  For example, a 0.10.0 broker can handle a v0, v1, or v2 FetchRequest.  On the other hand, this broker can only handle a v1 shutdown request, not a v0.  The response message that the broker sends will have the same version as the request that was sent.
 
-<p>Currently all versions are baselined at 0, as we evolve these APIs we will indicate the format for each version individually.</p>
+<p>Starting in 0.10.2, clients can also handle a range of versions.  This was implemented as part of KIP-97.  As the first thing it does, the client sends an ApiVersionsRequest to the broker to learn which versions of each API are supported.  The client will attempt to send a request that is supported by both itself and the broker.
```

Comment:
Agree. I liked starting with "The protocol is designed to..." - explaining what we were trying to do before diving into details of how we do things.